tools/c7n_azure/c7n_azure/resources/apimanagement.py

```diff
@@ -45,3 +48,57 @@ class resource_type(ArmResourceManager.resource_type):
             'resourceGroup'
         )
         resource_type = 'Microsoft.ApiManagement/service'
+
+
+@ApiManagement.action_registry.register('resize')
+class Resize(AzureBaseAction):
+    """
+    Action to scale api management resource.
+    Required arguments: capacity in units and tier (Developer, Basic, Standard or Premium).
+
+    :example:
+    This policy will resize api management to Premium tier with 8 units.
+
+    policies:
+      - name: resize-api
+        resource: azure.api-management
+        filters:
+          - type: value
+            key: name
+            value: test-api
+        actions:
+          - type: resize
+            tier: Premium
+            capacity: 8
+
+    """
+
+    schema = type_schema(
+        'resize',
+        required=['capacity', 'tier'],
+        **{
+            'capacity': {'type': 'number'},
+            'tier': {'enum': ['Developer', 'Basic', 'Standard', 'Premium']}

```

Comment:
I think these are actually queryable with sdk, so you could do validation checks on the policy oppose to using enum.

https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/apimanagement/azure-mgmt-apimanagement/azure/mgmt/apimanagement/operations/api_management_service_skus_operations.py

Reply:
Downside with looking it up is that we lose validation since validation happens without any access to Azure/credentials, so it becomes a run time error instead.  Although not having to maintain the list would be nice.

Reply:
Yeah, I was thinking similar way to Stefan. Talked to Andy offline and decided to leave as enum.